accounts/store_sql.go

```diff
@@ -0,0 +1,704 @@
+			return 0, err
+		}
+
+		numTries--
+	}
+
+	return 0, fmt.Errorf("couldn't create new account ID")
+}
+
+// AddAccountInvoice adds and invoice hash to the account with the given
+// AccountID alias.
+//
+// NOTE: This is part of the Store interface.
+func (s *SQLStore) AddAccountInvoice(ctx context.Context, alias AccountID,
+	hash lntypes.Hash) error {
+
+	var writeTxOpts db.QueriesTxOptions
+	return s.db.ExecTx(ctx, &writeTxOpts, func(db SQLQueries) error {
+		acctID, err := getAccountIDByAlias(ctx, db, alias)
+		if err != nil {
+			return err
+		}
+
+		// First check that this invoice does not already exist.
+		_, err = db.GetAccountInvoice(ctx, sqlc.GetAccountInvoiceParams{
+			AccountID: acctID,
+			Hash:      hash[:],
+		})
+		// If it does, there is nothing left to do.
+		if err == nil {
+			return nil
+		} else if err != nil && !errors.Is(err, sql.ErrNoRows) {
+			return err
+		}
+
+		err = db.AddAccountInvoice(ctx, sqlc.AddAccountInvoiceParams{
+			AccountID: acctID,
+			Hash:      hash[:],
+		})
+		if err != nil {
+			return err
+		}
+
+		return s.markAccountUpdated(ctx, db, acctID)
+	})
+}
+
+func getAccountIDByAlias(ctx context.Context, db SQLQueries, alias AccountID) (
+	int64, error) {
+
+	aliasInt, err := alias.ToInt64()
+	if err != nil {
+		return 0, fmt.Errorf("error converting account alias into "+
+			"int64: %w", err)
+	}
+
+	acctID, err := db.GetAccountIDByAlias(ctx, aliasInt)
+	if errors.Is(err, sql.ErrNoRows) {
+		return 0, ErrAccNotFound
+	}
+
+	return acctID, err
+}
+
+// markAccountUpdated is a helper that updates the last updated timestamp of
+// the account with the given ID.
+func (s *SQLStore) markAccountUpdated(ctx context.Context,
+	db SQLQueries, id int64) error {
+
+	_, err := db.UpdateAccountLastUpdate(
+		ctx, sqlc.UpdateAccountLastUpdateParams{
+			ID:          id,
+			LastUpdated: s.clock.Now().UTC(),
+		},
+	)
+	if errors.Is(err, sql.ErrNoRows) {
+		return ErrAccNotFound
+	}
+
+	return err
+}
+
+// UpdateAccountBalanceAndExpiry updates the balance and/or expiry of an
+// account.
+//
+// NOTE: This is part of the Store interface.
+func (s *SQLStore) UpdateAccountBalanceAndExpiry(ctx context.Context,
+	alias AccountID, newBalance fn.Option[int64],
+	newExpiry fn.Option[time.Time]) error {
+
+	var (
+		writeTxOpts db.QueriesTxOptions
+	)
+	return s.db.ExecTx(ctx, &writeTxOpts, func(db SQLQueries) error {
+		id, err := getAccountIDByAlias(ctx, db, alias)
+		if err != nil {
+			return err
+		}
+
+		newBalance.WhenSome(func(i int64) {
+			_, err = db.UpdateAccountBalance(
+				ctx, sqlc.UpdateAccountBalanceParams{
+					ID:                 id,
+					CurrentBalanceMsat: i,
+				},
+			)
+		})
+		if err != nil {
+			return err
+		}
+
+		newExpiry.WhenSome(func(t time.Time) {
+			_, err = db.UpdateAccountExpiry(
+				ctx, sqlc.UpdateAccountExpiryParams{
+					ID:         id,
+					Expiration: t.UTC(),
+				},
+			)
+		})
+		if err != nil {
+			return err
+		}
+
+		return s.markAccountUpdated(ctx, db, id)
```

Comment:
Should maybe check that either `newBalance` or `newExpiry` is `Some` before `markAccountUpdated`?

Reply:
https://github.com/lightninglabs/lightning-terminal/blob/1ba09c3cf0701bd50b41462f652846151de9c73a/accounts/store_kvdb.go#L193-L208

Reply:
Right, this is existing behaviour :)!
Also rechecked the old code prior to the addition of the `kvdb` refactors, and verified that it was existing behaviour prior to that as well.